src/transport/net.cu

```diff
@@ -152,12 +152,15 @@ static inline int groupBestEnd(int nranks, int* groups, int group, int* subgroup
 
 ncclResult_t netGetRings(int nranks, int* groups, int* subgroups, ncclTvalue_t* values, int* nringsRet, int* prev, int* next, int minScore, int* nthreads) {
   int nGroups = groups[nranks-1] + 1;
-  int cardUsed[NET_MAX_IFS*nGroups];
+  int* cardUsed;
+  NCCLCHECK(ncclCalloc(&cardUsed, NET_MAX_IFS*nGroups));
   for (int c=0; c<NET_MAX_IFS*nGroups; c++) cardUsed[c] = 0;
 
   for (int ring = 0; ring<*nringsRet; ring++) {
-    int starts[nGroups];
-    int ends[nGroups];
+    int* starts;
+    NCCLCHECK(ncclCalloc(&starts, nGroups));
+    int* ends;
+    NCCLCHECK(ncclCalloc(&ends, nGroups));
     for (int group = 0; group<nGroups; group++) {
```

Comment:
Maybe we should get those two allocations out of the loop to avoid multiple malloc/free.

Reply:
Done.